src/client/pydaos/daosdbm.py

```diff
@@ -45,9 +45,10 @@ def __init__(self, transport, interface, puid, cuid):
         except Exception:
             pass
 
-        os.environ['CRT_PHY_ADDR_STR'] = transport
-        os.environ['OFI_INTERFACE'] = interface
-        os.environ['DAOS_SINGLETON_CLI'] = '1'
+        if 'CRT_PHY_ADDR_STR' not in os.environ:
+            os.environ['CRT_PHY_ADDR_STR'] = transport
+        if interface and 'OFI_INTERFACE' not in os.environ:
+            os.environ['OFI_INTERFACE'] = interface
```

Comment:
I think that this should be provided by the environment (and eventually transparently via the local agent) and should be removed from the wrapper/API.